### PR TITLE
add Korean localization by JoungKyun Kim

### PR DESCRIPTION
**Pull request by Joungkyun**
Add Korean localization file. (openvpn-gui-res-kr.rc).
This file is written by UTF-8.

Thanks.

### CODE REVIEW

**Comment by mattock** [comment]
Thanks! Technically the changes look fine, but I'll try to get somebody to review the actual translation.

File: res/openvpn-gui-res-kr.rc [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+    /* Logviewer - Resources */
+    IDS_ERR_START_LOG_VIEWER "로그 뷰어 시작 실패: %s"
+    IDS_ERR_START_CONF_EDITOR "설정 편집기 시작 실패: %s"
+
+    /* OpenVPN */
+    IDS_ERR_MANY_CONFIGS "OpenVPN GUI는 %d config 이상의 설정은 지원하지 않습니다. 더이상 설정이 필요한 경우 개발자에게 연락 하십시오."
+    IDS_NFO_NO_CONFIGS "설정 파일이 없습니다.\n"\
+                       """파일 불러오기.."" 메뉴를 이용하거나 설정 파일을 ""%s"" 또는 ""%s""로 복사 하십시오."
+    IDS_ERR_CONFIG_NOT_AUTHORIZED "이 연결(%s)을 시작하기 위하여 ""%s"" 그룹 권한이 필요합니다. \"
```
Comment (selvanair) [inline res/openvpn-gui-res-kr.rc]: \\" at the end of this line should be "\\
Wonder why travis didn't catch this..

File: res/openvpn-gui-res-kr.rc [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+                                  "시스템 관리자에게 문의 하십시오.\n"
+    IDS_ERR_CONFIG_TRY_AUTHORIZE  "이 연결(%s)을 시작하기 위하여 ""%s"" 그룹 권한이 필요합니다.\n\n"\
+                                  "자신을 이 그룹에 추가하겠습니까?\n"\
+                                  "이 작업은 관리자 암호 또는 동의를 요구할 수 있습니다."
+    IDS_NFO_CONFIG_AUTH_PENDING   "이 연결(%s)을 시작하기 위하여 ""%s"" 그룹 권한이 필요합니다.\n\n"\
+                                  "이전의 인증 대화 상자를 완료 하십시오."
+    IDS_ERR_ADD_USER_TO_ADMIN_GROUP """%s"" group에 사용자 추가를 실패했습니다."
+    IDS_ERR_ONE_CONN_OLD_VER "OpenVPN 2.0-beta6 이전 버전에서는 2개 이상의 동시 연결을 지원하지 않습니다."
+    IDS_ERR_STOP_SERV_OLD_VER "You cannot use OpenVPN GUI to start a connection while the OpenVPN Service is running (with OpenVPN 1.5/1.6). Stop OpenVPN Service first if you want to use OpenVPN GUI."
```
Comment (selvanair) [inline res/openvpn-gui-res-kr.rc]: Missed or deliberately left out? There are a number of such below as well. 
Not everything need to be translated in one attempt, so its perfectly fine if you want to leave these for some other time.

File: res/openvpn-gui-res-kr.rc (hunk 1 above)
Comment (chipitsine) [inline res/openvpn-gui-res-kr.rc]: it "did" catch

https://travis-ci.org/OpenVPN/openvpn-gui/jobs/221950708#L683-L686

**Comment by selvanair** [comment]
On Sat, Apr 15, 2017 at 9:34 PM, Ilya Shipitsin <notifications@github.com>
wrote:

> +    IDS_NFO_NO_CONFIGS "설정 파일이 없습니다.\n"\
> +                       """파일 불러오기.."" 메뉴를 이용하거나 설정 파일을 ""%s"" 또는 ""%s""로 복사 하십시오."
> +    IDS_ERR_CONFIG_NOT_AUTHORIZED "이 연결(%s)을 시작하기 위하여 ""%s"" 그룹 권한이 필요합니다. \"
>
> it "did" catch
>
> https://travis-ci.org/OpenVPN/openvpn-gui/jobs/221950708#L683-L686
>
Aha, there is a warning indeed. I expected an error as lack of continuation
character leaves the next line dangling in the resource file. Warning is
better than nothing though..

**Comment by chipitsine** [comment]
people do not pay attention to warnings.
there was a "warning" on duplicate resources

https://github.com/OpenVPN/openvpn-gui/commit/727eaee0c54b0aee01ad28f622c1da7011e7b7c1

actually, it was a bug.

I couldn't find a way to turn those "warnings" into errors.

File: res/openvpn-gui-res-kr.rc (hunk 1 above)
Comment (Joungkyun) [inline res/openvpn-gui-res-kr.rc]: This problem was corrected to 8efb54a. please check.

File: res/openvpn-gui-res-kr.rc (hunk 2 above)
Comment (Joungkyun) [inline res/openvpn-gui-res-kr.rc]: IDS_ERR_STOP_SERV_OLD_VER is duplicated. In 45bdd20, remove english message.

File: res/openvpn-gui-res-kr.rc [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+
+/* Auth Username/Password Dialog */
+ID_DLG_AUTH DIALOG 6, 18, 160, 78
+STYLE WS_POPUP | WS_VISIBLE | WS_CAPTION | DS_CENTER | DS_SETFOREGROUND
+CAPTION "OpenVPN - 사용자 인증"
+FONT 8, "Microsoft Sans Serif"
+LANGUAGE LANG_ENGLISH, SUBLANG_DEFAULT
+BEGIN
+    LTEXT "사용자:", 0, 6, 9, 50, 10
+    LTEXT "암호:", 0, 6, 26, 50, 10
```
Comment (Joungkyun) [inline res/openvpn-gui-res-kr.rc]: As 4d97447, change the font name to the default font of Windows KR(Malgun Gothic) and increased font size to 9. And some fixed text positions. Plez check.

Comment (Joungkyun) [inline res/openvpn-gui-res-kr.rc]: login
config
config-proxy
config-detail
config-info
config-view-status

**Comment by selvanair** [comment]
Thanks for fixes.  Now the font and size looks good for hangul text. The backslashes showing up as the symbol for Korean Won is normal I suppose -- is it possible to get it displayed as a normal backslash?

Let's wait a bit more and see if anyone can review the translations.

**Comment by Joungkyun** [comment]
for displayed as normal backslash, we can change to the other fonts that displayed as normal backslash. However, it is not supported in Windows default fonts.

P.S.
I was the commiter of Mozilla L10n:ko, and I was the owner of thunderbird L10n:ko. I have been involved in translating Firefox 3.0 and Thunderbird 1.0 ~ 2.0. Currently, I'm registered as Alumni members on [
https://wiki.mozilla.org/L10n:Teams:ko](
https://wiki.mozilla.org/L10n:Teams:ko). So translation quality will not be bad

mozilla-l10n-ko

**Comment by selvanair** [comment]
Then leave the backslash as is -- most people are used to seeing it as "Won" I guess. 

> So translation quality will not be bad

I do not doubt it a bit.. We just have the policy of getting a review and ACK for all PRs as far as possible. If no review is forthcoming we'll lazy ACK this.

Again, thanks a lot for the translations --- this is a whole new language addition and a lot of work. Much appreciated.

**Comment by Joungkyun** [comment]
thank you, selvanair

**Comment by selvanair** [comment]
I think we should get this merged.
@Joungkyun Could you please squash the commits into two (the font change may be kept as one commit and all the rest as one) and force push?

**Comment by Joungkyun** [comment]
@selvanair Sorry, I am not familiar with git yet.

**Comment by selvanair** [comment]
@Joungkyun : I can squash and merge if its ok with you. It will look like this [branch](https://github.com/selvanair/openvpn-gui/commits/JK-squashed)

**Comment by Joungkyun** [comment]
I rebased and I reduced my commit to 3 ea. The last commit (fixed typo) was left in a non-contiguous sequence with the first commit. Is this correct?

**Comment by selvanair** [comment]
This looks good. Merging

**Pull request merged**